internal/zinc-persist/src/main/protobuf/schema.proto

```diff
@@ -0,0 +1,444 @@
+syntax = "proto3";
+
+package sbt.internal.inc;
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////// ANALYSIS API ////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+message Hash {
+    string hash = 1;
+}
+
+message LastModified {
+    int64 millis = 1;
+}
+
+message Stamps {
+    message StampType {
+        oneof type {
+            Hash hash = 1;
+            LastModified lastModified = 2;
+        }
+    }
+
+    map<string, StampType> productStamps = 1;
+    map<string, StampType> binaryStamps = 2;
+    map<string, StampType> sourceStamps = 3;
+}
+
+message OutputGroup {
+    string source = 1;
+    string target = 2;
+}
+
+message MultipleOutput {
+    repeated OutputGroup outputGroups = 1;
+}
+
+message SingleOutput {
+    string target = 1;
+}
+
+message Compilation {
+    int64 startTime = 1;
+    oneof Output {
+        SingleOutput singleOutput = 2;
+        MultipleOutput multipleOutput = 3;
+    }
+}
+
+message Compilations {
+    repeated Compilation compilations = 1;
+}
+
+enum Severity {
+    INFO = 0;
+    WARN = 1;
+    ERROR = 2;
+}
+
+message Position {
+    sint32 line = 1;
+    sint32 offset = 2;
+    string lineContent = 3;
+    sint32 pointer = 4;
+    string pointerSpace = 5;
+    string sourcePath = 6;
+    string sourceFilepath = 7;
+}
+
+message Problem {
+    string category = 1;
+    Severity severity = 2;
+    string message = 3;
+    Position position = 4;
+}
+
+message SourceInfo {
+    repeated Problem reportedProblems = 1;
+    repeated Problem unreportedProblems = 2;
+    repeated string mainClasses = 3;
+}
+
+message SourceInfos {
+    map<string, SourceInfo> sourceInfos = 1;
+}
+
+message FileHash {
+    string path = 1;
+    sint32 hash = 2;
+}
+
+enum CompileOrder {
+    MIXED = 0;
+    JAVATHENSCALA = 1;
+    SCALATHENJAVA = 2;
+}
+
+message Tuple {
+    string first = 1;
+    string second = 2;
+}
+
+message MiniOptions {
+    repeated FileHash classpathHash = 1;
+    repeated string scalacOptions = 2;
```

Comment:
Actually, you cannot use `Array` directly with scalapb, it defaults to `Seq`... So this change is not possible.